src/services/submissionService.ts

```diff
@@ -377,6 +386,142 @@ function validateIdInput(id: string): Result<unknown> {
     return validationResults
 }
 
+function validateSubmissionSearchFilters(filters: gqlTypes.SubmissionSearchFilters): Result<unknown> {
+    const validateSearchResults: Result<unknown> = {
+        data: undefined,
+        hasErrors: false,
+        errors: []
+    }
+
+    if (filters.orderBy?.some(order => order.fieldToOrder === 'id')) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.INVALID_ORDERBY_OPTION,
+            httpStatus: 400
+        })
+    }
+
+    const validOrderByFields = [
+        'googleMapsUrl',
+        'healthcareProfessionalName',
+        'isUnderReview',
+        'isApproved',
+        'isRejected',
+        'createdDate',
+        'updatedDate',
+]
+
+const invalidOrderByOption = filters.orderBy?.find(order => 
+    !validOrderByFields.includes(order.fieldToOrder))
+
+if (invalidOrderByOption) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_OPTION_NOT_FOUND,
+        httpStatus: 400
+    })
+}
+
+if (filters.orderBy && filters.orderBy.length > 2) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_OPTION_MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.orderBy && filters.orderBy.length === 2 && 
+    filters.orderBy[0].fieldToOrder === filters.orderBy[1].fieldToOrder) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_FIELD_SELECTED_TWICE,
+        httpStatus: 400
+    })
+}
+
+if (filters.limit && filters.limit > 1000) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'limit',
+        errorCode: ErrorCode.MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.limit && filters.limit < 0) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'limit',
+        errorCode: ErrorCode.MIN_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.offset && filters.offset > 10000) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'offset',
+        errorCode: ErrorCode.MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.offset && filters.offset < 0) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'offset',
+        errorCode: ErrorCode.MIN_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.googleMapsUrl === '' || (filters.googleMapsUrl && !filters.googleMapsUrl.trim())) {
+    validateSearchResults.hasErrors = true
```

Comment:
I tried adding the shorter line `if (!filters.googleMapsUrl || !filters.googleMapsUrl.trim())` suggested in a previous PR review, but it causes manual tests to fail whenever `googleMapsUrl` is not passed as a filter.